pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -174,6 +175,7 @@ protected ConsumerImpl(PulsarClientImpl client, String topic, ConsumerConfigurat
         this.priorityLevel = conf.getPriorityLevel();
         this.readCompacted = conf.isReadCompacted();
         this.subscriptionInitialPosition = conf.getSubscriptionInitialPosition();
+        this.negativeAcksTracker = new NegativeAcksTracker(this, conf);
```

Comment:
can we add `NoOpNegativeAcksTracker` if this feature is not required by the client.

Reply:
If a user doesn't call negativeAck there won't be much overhead anyway. Also, whether the feature is required by client, will be determined by the client calling the API or not.

Reply:
we could keep the set initialized to null until the first nack is called. That will bring the overhead down to the same level as a NoOpNegativeAcksTracker